### PR TITLE
OH2-252 | Fix audit logging from API

### PR DESCRIPTION
**Pull request by ArnaudFonzam**
See OH2-252.

### CODE REVIEW

File: src/main/java/org/isf/login/rest/LoginController.java [hunk 1]
```diff
@@ -77,11 +83,11 @@ public ResponseEntity<LoginResponse> authenticateUser(@Valid @RequestBody LoginR
 		String jwt = tokenProvider.generateJwtToken(authentication, true);
 
 		String userDetails = (String) authentication.getPrincipal();
-		MDC.put("OHUser", loginRequest.getUsername());
```
Comment (mwithi) [inline src/main/java/org/isf/login/rest/LoginController.java:80]: import in line 42 is no longer needed

File: src/main/java/org/isf/login/rest/LoginController.java [hunk 2]
```diff
@@ -77,11 +83,11 @@ public ResponseEntity<LoginResponse> authenticateUser(@Valid @RequestBody LoginR
...
-
 		try {
 			this.httpSession.setAttribute("sessionAuditId",
 							sessionAuditManager.newSessionAudit(new SessionAudit(userDetails, LocalDateTime.now(), null)));
+			User user = userManager.getUserByName(loginRequest.getUsername());
+			UserSession.setUser(user);
```
Comment (mwithi) [inline src/main/java/org/isf/login/rest/LoginController.java:90]: These should precede `newSessionAudit`

**Review by mwithi — CHANGES_REQUESTED** [review]
A number of changes requested on both CORE and API, and GUI too.
Apparently logout is not working from API.

File: src/main/java/org/isf/login/rest/LoginController.java (hunk 2 above)
Comment (ArnaudFonzam) [reply src/main/java/org/isf/login/rest/LoginController.java]: yes

File: src/main/java/org/isf/login/rest/LoginController.java [hunk 3]
```diff
@@ -77,8 +83,15 @@ public ResponseEntity<LoginResponse> authenticateUser(@Valid @RequestBody LoginR
...
-		MDC.put("OHUser", loginRequest.getUsername());
-
+		User user;
+		try {
+			user = userManager.getUserByName(loginRequest.getUsername());
+			UserSession.setUser(user);
+		} catch (OHServiceException e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
+		}
```
Comment (mwithi) [inline src/main/java/org/isf/login/rest/LoginController.java:93]: Why to have a separated try/catch block when we can use the same below?

File: src/main/java/org/isf/security/CustomLogoutHandler.java [hunk 4]
```diff
@@ -53,14 +52,15 @@ public class CustomLogoutHandler implements LogoutHandler {
 	@Override
 	public void logout(HttpServletRequest request, HttpServletResponse response, Authentication authentication) {
 		try {
-			Optional<Object> sessionAuditIdOpt = Optional.ofNullable(httpSession.getAttribute("sessionAuditId"));
-			if (sessionAuditIdOpt.isPresent()) {
-				int sessionAuditId = (int) sessionAuditIdOpt.get();
+			int sessionAuditId = UserSession.getSessionAuditId();
+			System.out.println("sessionAudit :"+sessionAuditId);
```
Comment (mwithi) [inline src/main/java/org/isf/security/CustomLogoutHandler.java:56]: change it to LOGGER or remove println

**Review by mwithi — APPROVED** [review]

File: src/main/java/org/isf/security/ApiAuditorAwareImpl.java [hunk 5]
```diff
@@ -0,0 +1,21 @@
+package org.isf.security;
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/security/ApiAuditorAwareImpl.java:1]: Missing license header

**Review by dbmalkovsky — APPROVED** [review]
LGTM; after adding the missing license header

File: src/main/java/org/isf/security/ApiAuditorAwareImpl.java (hunk 5 above)
Comment (mwithi) [reply src/main/java/org/isf/security/ApiAuditorAwareImpl.java:22]: Done, thanks!

**Comment by mwithi** [comment]
Checks fail for a "not-really" breaking change in the oh.yaml specification, will investigate later

**Pull request merged**